core/src/main/scala/shapeless/typeable.scala

```diff
@@ -139,15 +157,17 @@ object Typeable extends TupleTypeableInstances with LowPriorityTypeable {
           else for(e <- o; _ <- e.cast[T]) yield t.asInstanceOf[Option[T]]
         } else None
       }
+      def describe = s"Option[${castT.describe}]"
     }
 
   /** Typeable instance for `Either`. */
   implicit def eitherTypeable[A, B]
-    (implicit castA: Typeable[Left[A, B]], castB: Typeable[Right[A, B]]): Typeable[Either[A, B]] =
+    (implicit castA: Typeable[Left[A, B]], castB: Typeable[Right[A, B]], castC: Typeable[A], castD: Typeable[B]): Typeable[Either[A, B]] =
```

Comment:
When using just castA and castB, the output for Either[Long, String] is something like "Typeable[Either[Left[Long], Right[String]]]", that is I why I added castC and castD, However, if we remove castA and castB, it will fail to require that arguments are Either in the the first place, won't it?